### PR TITLE
Intrepid2 - uvm removal testing on shared utils, integration, C1 basis

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
## Motivation

- Testing without uvm. This PR is simple changing cuda execution space with device<cuda,cudaspace>.

- Now it is a bit more complicated PR. I converted Integration code without UVM. A few work items are found.

- C1 basis functions are converted. 

## Testing

Tests are passed on weaver V100.

### CODE REVIEW

**Comment by kyungjoo-kim** [comment]
https://github.com/trilinos/Trilinos/issues/8310

cross reference

File: packages/intrepid2/unit-test/Discretization/Integration/test_05.hpp [hunk 1]
```diff
@@ -139,16 +141,16 @@ namespace Intrepid2 {
 
       // compute and compare integrals
       try {
-        const auto maxDeg   = Parameters::MaxCubatureDegreeTri;
+        const auto maxDeg   = 10; //Parameters::MaxCubatureDegreeTri;
```
Comment (mperego) [inline packages/intrepid2/unit-test/Discretization/Integration/test_05.hpp:144]: Why did you change this?

File: packages/intrepid2/unit-test/Discretization/Integration/test_06.hpp [hunk 2]
```diff
@@ -141,17 +143,17 @@ namespace Intrepid2 {
       // compute and compare integrals
       try {
         // cannot test maxcubature degree edge (20) as max integration point is limited by 1001.
-        const auto maxDeg   = Parameters::MaxCubatureDegreeTet;
+        const auto maxDeg   = 10; //Parameters::MaxCubatureDegreeTet;
```
Comment (mperego) [inline packages/intrepid2/unit-test/Discretization/Integration/test_06.hpp:146]: sam here

File: packages/intrepid2/unit-test/Discretization/Integration/test_07.hpp [hunk 3]
```diff
@@ -145,17 +147,17 @@ namespace Intrepid2 {
       // compute and compare integrals
       try {
         // cannot test maxcubature degree edge (20) as max integration point is limited by 1001.
-        const auto maxDeg   = Parameters::MaxCubatureDegreePyr;
+        const auto maxDeg   = 10; //Parameters::MaxCubatureDegreePyr;
```
Comment (mperego) [inline packages/intrepid2/unit-test/Discretization/Integration/test_07.hpp:150]: and here.

**Review by mperego — APPROVED** [review]
Looks good! Thanks for doing this @kyungjoo-kim 
I only have one comment.

**Comment by mperego** [comment]
@kyungjoo-kim I removed the Automerge in case you want to address my comment.

File: packages/intrepid2/unit-test/Discretization/Integration/test_05.hpp (hunk 1 above)
Comment (kyungjoo-kim) [reply packages/intrepid2/unit-test/Discretization/Integration/test_05.hpp:144]: It takes too much time. Now we do not use UVM, all test verification includes explicit deep copy and a lot more kernels launches. This is all work in progress and we will consider to test all when we complete the uvm removal.

Comment (kyungjoo-kim) [reply packages/intrepid2/unit-test/Discretization/Integration/test_05.hpp:144]: In the next PR, I will only lower the order for cuda and host code will do test the max.
